src/aiida/engine/daemon/execmanager.py

```diff
@@ -373,7 +373,7 @@ async def _copy_local_files(logger, node, transport, inputs, local_copy_list, wo
                 # If the source object is a directory, we copy its entire contents
                 data_node.base.repository.copy_tree(filepath_target, filename_source)
                 await transport.put_async(
-                    f'{dirpath}/*',
```

Comment:
This will copy everything inside `dirpath` to `workdir`. If `filename_target` is a `out/HP`, then the `dirpath` will have `out/HP`, then it will copy them into the workdir target `out/HP`, result folders as reported in #6767